### PR TITLE
[SHELL32] Control panel context menu bugfix

### PR DESCRIPTION
**Pull request by nothotscott**
## Purpose

Fixes context menus not invoking commands when right clicking for Control panel items

## Proposed changes

Change the context menu ID offset

### CODE REVIEW

File: dll/win32/shell32/folders/CControlPanelFolder.cpp [hunk 1]
```diff
@@ -676,9 +676,9 @@ HRESULT WINAPI CCPLItemMenu::QueryContextMenu(
     UINT idCmdLast,
     UINT uFlags)
 {
-    _InsertMenuItemW(hMenu, indexMenu++, TRUE, IDS_OPEN, MFT_STRING, MAKEINTRESOURCEW(IDS_OPEN), MFS_DEFAULT);
-    _InsertMenuItemW(hMenu, indexMenu++, TRUE, idCmdFirst + 1, MFT_SEPARATOR, NULL, MFS_ENABLED);
-    _InsertMenuItemW(hMenu, indexMenu++, TRUE, IDS_CREATELINK, MFT_STRING, MAKEINTRESOURCEW(IDS_CREATELINK), MFS_ENABLED);
+    _InsertMenuItemW(hMenu, indexMenu++, TRUE, idCmdFirst + IDS_OPEN, MFT_STRING, MAKEINTRESOURCEW(IDS_OPEN), MFS_DEFAULT);
```
Comment (learn-more) [inline dll/win32/shell32/folders/CControlPanelFolder.cpp:679]: This should just be `idCmdFirst`

File: dll/win32/shell32/folders/CControlPanelFolder.cpp [hunk 2]
```diff
@@ -676,9 +676,9 @@ HRESULT WINAPI CCPLItemMenu::QueryContextMenu(
...
+    _InsertMenuItemW(hMenu, indexMenu++, TRUE, IDC_STATIC, MFT_SEPARATOR, NULL, MFS_ENABLED);
+    _InsertMenuItemW(hMenu, indexMenu++, TRUE, idCmdFirst + IDS_CREATELINK, MFT_STRING, MAKEINTRESOURCEW(IDS_CREATELINK), MFS_ENABLED);
```
Comment (learn-more) [inline dll/win32/shell32/folders/CControlPanelFolder.cpp:681]: This should be idCmdFirst + 1

File: dll/win32/shell32/folders/CControlPanelFolder.cpp (hunk 1 above)
Comment (nothotscott) [reply dll/win32/shell32/folders/CControlPanelFolder.cpp:679]: How should CCPLItemMenu::InvokeCommand identify the menu item then?

Comment (learn-more) [reply dll/win32/shell32/folders/CControlPanelFolder.cpp:679]: I am not sure actually, we have shell extensions that expect the the `MAKEINTRESOURCEW(IDS_OPEN)` to take care of that, others are manually adding stuff...

**Review by learn-more — APPROVED** [review]
lgtm

File: dll/win32/shell32/folders/CControlPanelFolder.cpp [hunk 3]
```diff
@@ -699,7 +699,7 @@ HRESULT WINAPI CCPLItemMenu::InvokeCommand(LPCMINVOKECOMMANDINFO lpcmi)
 
     TRACE("(%p)->(invcom=%p verb=%p wnd=%p)\n", this, lpcmi, lpcmi->lpVerb, lpcmi->hwnd);
 
-    if (lpcmi->lpVerb == MAKEINTRESOURCEA(IDS_OPEN)) //FIXME
+    if (lpcmi->lpVerb == MAKEINTRESOURCEA(0))
```
Comment (HBelusca) [inline dll/win32/shell32/folders/CControlPanelFolder.cpp:702]: this "0" (and "1" in the other part of the diff) appear at first sight "arbitrary"/"magic". What do they correspond to?

Comment (learn-more) [reply dll/win32/shell32/folders/CControlPanelFolder.cpp:702]: index 0 and 1 of the values added to the menu before?
As documented on MSDN:
https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-icontextmenu-invokecommand

Comment (HBelusca) [reply dll/win32/shell32/folders/CControlPanelFolder.cpp:702]: thanks for the link

**Pull request merged**